### PR TITLE
Add legalAcceptedAt and skipLegalChecks fields to user management methods

### PR DESCRIPTION
**Pull request by NWylynko**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1828/references/backend/user/get-user-list#examples
> - https://clerk.com/docs/pr/1828/references/backend/user/get-user#example
> - https://clerk.com/docs/pr/1828/references/backend/user/create-user#create-user-params
> - https://clerk.com/docs/pr/1828/references/backend/user/update-user#update-user-params
> - https://clerk.com/docs/pr/1828/references/backend/types/backend-user#properties

### Explanation:

Updates the user object to include `legalAcceptedAt` and `skipLegalChecks` in the backend sdk to reflect the backend api.

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

### CODE REVIEW

**Comment by royanger** [comment]
@jeremy-clerk, I added you for a review as you originally reported this.

**Review by royanger — COMMENTED** [review]
Does the return from `clerkClient.users.updateUser()` and `clerkClient.users.updateUser()` include `legalAcceptedAt`? I would expect it so we might need to update the `console.log(response)` on both of those pages.

**Comment by NWylynko** [comment]
> Does the return from `clerkClient.users.updateUser()` and `clerkClient.users.updateUser()` include `legalAcceptedAt`? I would expect it so we might need to update the `console.log(response)` on both of those pages.

Good catch! added :)

**Comment by alexisintech** [comment]
hi friend! thanks so much for getting on this ticket.
we will need to update this page as well:
- https://clerk.com/docs/references/backend/types/backend-user
  - relevant resource: https://github.com/clerk/javascript/blob/main/packages/backend/src/api/endpoints/UserApi.ts

File: docs/references/backend/user/create-user.mdx [hunk 1]
```diff
@@ -201,6 +201,20 @@ function createUser(params: CreateUserParams): Promise<User>
 
   ---
 
+  - `legalAcceptedAt?`
+  - `string`
```
Comment (alexisintech) [inline docs/references/backend/user/create-user.mdx:205]:
```suggestion
  - `Date`
```

Comment (alexisintech) [reply docs/references/backend/user/create-user.mdx:205]: the backend api lists it as string or null, but it looks like the backend sdk has it listed as a `Date`
https://github.com/clerk/javascript/blob/main/packages/backend/src/api/endpoints/UserApi.ts#L78

File: docs/references/backend/user/create-user.mdx [hunk 2]
```diff
@@ -201,6 +201,20 @@ function createUser(params: CreateUserParams): Promise<User>
...
+
+  A custom timestamp denoting when the user accepted legal requirements, specified in [RFC3339 format](https://datatracker.ietf.org/doc/html/rfc3339). For example: `2012-10-20T07:15:20.902Z`.
+
+  ---
+
+  - `skipLegalChecks?`
+  - `boolean`
+
+  When set to true all legal checks are skipped. It is not recommended to skip legal checks unless you are migrating a user to Clerk.
```
Comment (alexisintech) [inline docs/references/backend/user/create-user.mdx:214]:
```suggestion
  When set to `true`, all legal checks are skipped. It is not recommended to skip legal checks unless you are migrating a user to Clerk.
```

File: docs/references/backend/user/create-user.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/references/backend/user/create-user.mdx:205]: please apply this update to updateUser as well :)

File: docs/references/backend/user/create-user.mdx (hunk 2 above)
Comment (alexisintech) [reply docs/references/backend/user/create-user.mdx:214]: please apply this update to updateUser as well :)

**Comment by NWylynko** [comment]
Thanks for the feedback @alexisintech :) everything should be in order.

**Review by alexisintech — APPROVED** [review]
awesome, thanks so much 😸💖

**Pull request merged**